iam.tf

```diff
@@ -21,6 +21,23 @@ data "aws_iam_policy_document" "sns_feedback" {
   }
 }
 
+// See https://repost.aws/knowledge-center/monitor-sns-texts-cloudwatch for required permissions to deliver status logs
+data "aws_iam_policy_document" "sns_feedback_allow_delivery_status_logs" {
+  count = local.create_sns_feedback_role ? 1 : 0
+
+  statement {
+    effect = "Allow"
+    actions = [
+      "logs:CreateLogGroup",
+      "logs:CreateLogStream",
+      "logs:PutLogEvents",
+      "logs:PutMetricFilter",
+      "logs:PutRetentionPolicy",
+    ]
+    resources = ["*"]
```

Comment:
we know the ARN and can scope it with the `aws_cloudwatch_log_group` created by the module instead of using a wildcard

Reply:
I believe that is the log group for Lambda logs, which is not what we want here. We instead need the SNS feedback log groups, per my other comment. I could restrict it down to `sns/:region/:account/:topic` and `sns/:region/:account/:topic/*`? (I know the `/Failure` group is needed, but I'm not sure about others. The [reference post](https://repost.aws/knowledge-center/monitor-sns-texts-cloudwatch) just uses `*`, so that's not helpful)